src/fam_struct_wrapper.rs

```diff
@@ -0,0 +1,586 @@
+    // will actually be used as a `T`. The remaining memory in the `Vec<T>` is for `entries`,
+    // which must be contiguous. Since the entries are of type `FamStruct::Entry` we must
+    // be careful to convert the desired capacity of the `FamStructWrapper`
+    // from `FamStruct::Entry` to `T` when reserving or releasing memory.
+    mem_allocator: Vec<T>,
+    // the number of elements of type `FamStruct::Entry` currently in the vec
+    len: usize,
+    // the capacity of the `FamStructWrapper` measured in elements of type `FamStruct::Entry`
+    capacity: usize,
+}
+
+impl<T: Default + FamStruct> FamStructWrapper<T> {
+    /// Get the capacity required by mem_allocator in order to hold
+    /// the provided number of `FamStruct::Entry`
+    ///
+    fn mem_allocator_len(wrapper_len: usize) -> usize {
+        let wrapper_size_in_bytes = size_of::<T>() + wrapper_len * size_of::<T::Entry>();
+        (wrapper_size_in_bytes + size_of::<T>() - 1) / size_of::<T>()
+    }
+
+    /// Get the number of elements of type `FamStruct::Entry` that fit
+    /// in a mem_allocator of provided len
+    ///
+    fn fam_len(mem_allocator_len: usize) -> usize {
+        if mem_allocator_len == 0 {
+            return 0;
+        }
+
+        let array_size_in_bytes = (mem_allocator_len - 1) * size_of::<T>();
+        array_size_in_bytes / size_of::<T::Entry>()
+    }
+
+    /// Constructs a new FamStructWrapper<T> that contains `num_elements`
+    /// default-initialized elements of type `FamStruct::Entry`
+    ///
+    /// # Arguments
+    ///
+    /// * `num_elements` - The number of default-initialized elements of type
+    ///                    `FamStruct::Entry` in the initial `FamStructWrapper`
+    ///
+    pub fn new(num_elements: usize) -> FamStructWrapper<T> {
+        let required_mem_allocator_capacity =
+            FamStructWrapper::<T>::mem_allocator_len(num_elements);
+
+        let mut mem_allocator = Vec::with_capacity(required_mem_allocator_capacity);
+        for _ in 0..required_mem_allocator_capacity {
+            mem_allocator.push(T::default())
+        }
+        mem_allocator[0].set_len(num_elements);
+
+        FamStructWrapper {
+            mem_allocator,
+            len: num_elements,
+            capacity: num_elements,
+        }
+    }
+
+    /// Creates a new `FamStructWrapper` structure based on a supplied slice
+    /// of `FamStruct::Entry`.
+    ///
+    /// # Arguments
+    ///
+    /// * `entries` - The vector of `FamStruct::Entry` entries.
+    ///
+    pub fn from_entries(entries: &[T::Entry]) -> FamStructWrapper<T> {
+        let mut adapter = FamStructWrapper::<T>::new(entries.len());
+
+        {
+            let wrapper_entries = adapter.as_mut_fam_struct().as_mut_slice();
+            wrapper_entries.copy_from_slice(entries);
+        }
+
+        adapter
+    }
+
+    /// Get a reference to the actual `FamStruct` instance.
```

Comment:
```suggestion
    /// Get a reference to the actual [`FamStruct`](trait.FamStruct.html) instance.
```

This applies to all the following mentions too, I'll stop with the suggestions :smile:

Reply:
Fixed